bitcoin/src/light/wallet.rs

```diff
@@ -464,27 +451,26 @@ impl Wallet {
 
             // NOTE: we don't support signing p2sh, p2pkh, p2wsh inputs
             // since the Vault is assumed to only receive p2wpkh payments
-            let script_code = if prev_out.script_pubkey.is_v0_p2wpkh() {
-                Ok(p2wpkh_script_code(&prev_out.script_pubkey))
-            } else {
-                Err(Error::InvalidPrevOut)
-            }?;
+            let script_code = prev_out
+                .script_pubkey
+                .p2wpkh_script_code()
```

Comment:
Nice to have this implemented for us

Reply:
Yea the bitcoin crate has a lot of utilities, there is probably more code we can replace, but I wanted to keep the pr as minimal as possible